managed/services/management/service.go

```diff
@@ -162,3 +173,111 @@ func (s *ServiceService) validateRequest(request *managementpb.RemoveServiceRequ
 	}
 	return nil
 }
+
+// ListServices returns a filtered list of Services with some attributes from Agents and Nodes.
+//
+//nolint:unparam
+func (s *ServiceService) ListServices(ctx context.Context, req *managementpb.ListServiceRequest) (*managementpb.ListServiceResponse, error) {
+	filters := models.ServiceFilters{
+		NodeID:        req.NodeId,
+		ServiceType:   convertServiceType(req.ServiceType),
+		ExternalGroup: req.ExternalGroup,
+	}
+
+	var services []*models.Service
+	var agents []*models.Agent
+	var nodes []*models.Node
+
+	agentToAPI := func(agent *models.Agent) *managementpb.UniversalAgent {
+		return &managementpb.UniversalAgent{
+			AgentId:     agent.AgentID,
+			AgentType:   string(agent.AgentType),
+			Status:      agent.Status,
+			IsConnected: s.r.IsConnected(agent.AgentID),
+		}
+	}
+
+	// TODO: provide a higher level of data consistency guarantee by using a locking mechanism.
+	errTX := s.db.InTransaction(func(tx *reform.TX) error {
+		var err error
+		services, err = models.FindServices(tx.Querier, filters)
+		if err != nil {
+			return err
+		}
+
+		agents, err = models.FindAgents(tx.Querier, models.AgentFilters{})
+		if err != nil {
+			return err
+		}
+
+		nodes, err = models.FindNodes(tx.Querier, models.NodeFilters{})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if errTX != nil {
+		return nil, errTX
+	}
+
+	nodeMap := make(map[string]string, len(nodes))
+	for _, node := range nodes {
+		nodeMap[node.NodeID] = node.NodeName
+	}
+
+	resultSvc := make([]*managementpb.UniversalService, 0, len(services))
+	for i, service := range services {
+		labels, err := service.GetCustomLabels()
+		if err != nil {
+			return nil, err
+		}
+
+		svc := &managementpb.UniversalService{
+			Address:        pointer.GetString(service.Address),
+			Agents:         []*managementpb.UniversalAgent{},
```

Comment:
if we return agents with services, what is the reason to have a separate endpoint for agents?

Reply:
> if we return agents with services, what is the reason to have a separate endpoint for agents?

They deliver data for two different use cases.

Reply:
P.2 of [this ticket](https://jira.percona.com/browse/PMM-8234) talks about providing a list of agents related to a particular service. Every item should be expandable so the User can see all of the attributes.

<img width="906" alt="Screenshot 2023-04-03 at 07 00 13" src="https://user-images.githubusercontent.com/81549/229408946-ab75d04a-58a8-4fc1-a67a-707342a8eb25.png">